### DISCUSSION TITLE
Onlyoffice Client error

### DISCUSSION THREAD

**Discussion by jvelarde-mpi** [ONLYOFFICE/DocumentServer#1997] [Q&A]
Hello,
I have a owncloud and onlyoffice live and running, both on Ubunut 22.04. On the owncloud I enable Onlyoffice from the Market. I  input the ONLYOFFICE Docs address, but I get the following error:

Error when trying to connect (Client error response [url] https://onlyoffice.xxxxxxxxx.net/welcome/healthcheck [status code] 404 [reason phrase] Not Found)

Any help is appreciated, thanks.

**Comment by ShockwaveNN** [comment]
Did you tried to open https://onlyoffice.xxxxxxxxx.net/welcome/healthcheck in your browser? It should return page with true value if DocumentServer running fine

**Reply by jvelarde-mpi** [reply]
I get:

Cannot GET /welcome/healthcheck

**Reply by ShockwaveNN** [reply]
Something wrong with your setup

Describe how exactly you setup ONLYOFFICE DocumentServer?

**Reply by jvelarde-mpi** [reply]
I followed these steps
https://ubuntu.com/tutorials/how-to-install-onlyoffice-server-on-ubuntu#2-installation

Then enabled SSL:
https://helpcenter.onlyoffice.com/installation/docs-community-https-linux.aspx#:~:text=The%20easiest%20way%20to%20switch,Encrypt%20SSL%20Certificates%20using%20certbot.&text=The%20script%20will%20automatically%20create,the%20changes%20to%20take%20effect.

**Reply by jvelarde-mpi** [reply]
I found a few "tricks" so far the error message changed, please see attachment.
![Screenshot_2022-11-09-23-33-08-453](https://user-images.githubusercontent.com/117692015/201028293-9593f064-4187-4f51-8404-948fb63463e1.jpg)

**Reply by ShockwaveNN** [reply]
Is it working with disabled https?

Let's rule out netxcloud from it
Just check `/healthcheck` page

**Reply by jvelarde-mpi** [reply]
Am using hyper-v with checkpoints, I'll restore it to a fresh install with no SSL enable and test it out. I'll get back to you.

**Reply by ShockwaveNN** [reply]
Usually those strange error means that something wrong with SSL setup and maybe there are errors in nginx logs

**Reply by jvelarde-mpi** [reply]
I did a fresh install of onlyoffice on ubuntu 22.04 now the health check works
https://onlyoffice.modernparking.net/healthcheck it shows "true"

and Document server is running fine.
https://onlyoffice.modernparking.net/welcome/

Now when am trying to integrated in owncloud I get a different error, please see attachment, any ideas??

![error](https://user-images.githubusercontent.com/117692015/201168499-b2f0ff40-6b4c-4408-b862-c754da85fbe2.jpg)

**Reply by jvelarde-mpi** [reply]
After hours of searching online I found the fix;
On the onlyoffice change the default.json and local.json located in: (/var/etc/onlyoffice/documentserver/) to following

default.json search for "rejectUnauthorized" and change it false.
local.json search for "header", their should be 2 and change it to "AuthorizationJwt" 
Then run supervisor restart all

On the owncloud add the following lines to the config.php, located in /var/www/owncloud/config/

‘onlyoffice’ => array (
‘verify_peer_off’ => true,
‘jwt_header’ => “AuthorizationJwt”
)

Went back to my owncloud and integrated my OnlyOffice with the Secret Key, which you can get by running sudo documentserver-jwt-status.sh on the onlyoffice server.
